docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+
+ - `prepare()` - Sets `isMod` flag according to user's 
+ - `act()`- Gets `targetList` and calculates the raw ratio of the completed items.
+ - `roundedRatioBar()`- Returns a rounded ratio enclosed in square brackets for printing. 
+
+Given below is an example usage scenario and how the statistic mechanism behaves at each step. 
+
+Step 1. The user enters `stats -mod CS2113`	once the execute layer executes the message and calls `action.prepare()` class, `StatsAction` will begin its `prepare()` operation
+
+Step 2. `prepare()` looks at the input called `ParamNode args` which is user command processed by Command Intepreter layer, and starts to identify whether user has enter the keyword `mod ` if `userInput` contains the keyword, then `isMod` flag will be set. 
+
+Step 3. Next, execute layer will call `action.act()` which causes StatsAction to begin its `act()` operation. If `isMod` flag is set, `act()` will search for the user specified module and get the list of tasks tagged to it.
+
+Step 5. Once the list of task is obtain, the operation will loop through the task list and count the number of completed task followed by generating a ratio. 
+
+Step 6. This ratio will be passed into `roundedRatioBar` to return *String* of a rounded ratio to 1 decimal place enclosing it in square brackets. 
+
+Step 7. Now `StatsAction` is completed and it will return this string back to `Execute` for to be printed through `UI`. 
+
+
+**Design consideration:**
+
+**Aspect : How statistics executes**
+ - **Alternative 1 (current choice):** Create a separate class and get list of tasks/taken modules' task and scan through them to calculate statistics
+	 - Pros: Reduces Coupling and increase testability as a software unit itself. 
+	 - Cons: May have performance issues in terms of memory usage 
+
+ - **Alternative 2:** initialize statistics as zero and each task contains an aspect called statistics
+	 - Pros: Will use less memory since the task itself will be deleted. 
+	 - Cons: Stats will be updated constantly even though we do not need it. 
+
+### 4.3 Checker Feature 
+
+The checker mechanism is facilitated by the utility class `Checker`. It is an independent class on its own without extensions and is stored under the `Data` package of our app. The class implements the following operations: 
+
+ - `checkDuplicates()`- Calls the checkClash method and return the status of boolean variable `isClash` .
+ - `checkClash(ArrayList< item >, Item)`- Updates `isClash` once a duplicate item is found in the list.
+ - `checkRecurrenceDate(Task)` - Checks if the current date is beyond the stated date in the list, and provides a new update for the date recurring date.
+
+Given below is an example usage scenario and how the checker mechanism behaves at each step. 
+
+![here](Images/Checker_Diagram.png)
```

Comment:
Could include activation bars in your sequence diagram.
Otherwise, good job!
![image](https://user-images.githubusercontent.com/60440731/97668261-8d327480-1abc-11eb-9993-2743a6e056c4.png)